File: code/src/sneer/bricks/hardware/io/log/impl/LoggerImpl.scala
package sneer.bricks.hardware.io.log.impl

import sneer.bricks.hardware.io.log.Logger
import sneer.bricks.hardware.io.log.worker.LogWorker
import sneer.bricks.hardware.io.log.worker.LogWorkerHolder
import sneer.foundation.environments.Environments.my

class ScalaLoggerImpl extends Logger {

	var _delegate: LogWorker = null;
 
	override def log(message: String, messageInsets: AnyRef*): Unit = {
		if (_delegate == null) {
			_delegate = my(classOf[LogWorkerHolder]).worker();
			if (_delegate == null) return;
		}
		
		_delegate.log(message, messageInsets);
	}

}

File: code/src/sneer/bricks/hardware/io/log/impl/LoggerImpl.java
package sneer.bricks.hardware.io.log.impl;

import sneer.bricks.hardware.io.log.Logger;
import sneer.bricks.hardware.io.log.worker.LogWorker;
import sneer.bricks.hardware.io.log.worker.LogWorkerHolder;
import static basis.environments.Environments.my;

class LoggerImpl implements Logger {

	private LogWorker _delegate;

	
	@Override
	public void log(String message, Object... messageInsets) {
		if (_delegate == null) {
			_delegate = my(LogWorkerHolder.class).worker();
			if (_delegate == null) return;
		}
		
		_delegate.log(message, messageInsets);
	}

}